app/commands/solution/search_community_solutions.rb

```diff
@@ -68,28 +72,84 @@ def search_query
           must: [
             { term: { 'exercise.id': exercise.id } },
             { term: { status: 'published' } },
-            @criteria.blank? ? nil : { wildcard: { 'user.handle': "*#{criteria}*" } }
+            @up_to_date.nil? ? nil : { term: { 'out_of_date': !@up_to_date } },
```

Comment:
This is sort of a four-way boolean. We have:
1. show up to date only
2. show not up to date
3. show everything
4. Nothing passed in (so use default - probably 3)

Thoughts on how best to handle this?

Reply:
I'd argue it's a three way boolean, which option 3 and 4 being the same. I'm not a big fan of boolean arguments myself, but I could not come up with a good name that allowed me to pass in keywords. Thoughts?